src/artm/core/instance.cc

```diff
@@ -372,6 +384,11 @@ void Instance::DisposeRegularizer(const std::string& name) {
   schema_.set(new_schema);
 }
 
+void Instance::CreateOrReconfigureDictionaryImpl(const DictionaryData& data) {
```

Comment:
This method (`CreateOrReconfigureDictionaryImpl`) should create dictionary in two places -- both `dictionaries_impl_` and `dictionaries_`. Here is the right place to call a method that creates an old dictionary (`Dictionary`) from `DictionaryData`. This is a temporary hack that will ensure that all scores and regularizers still work if user creates a new dictionary.